pkg/util/net/http.go

```diff
@@ -67,6 +70,9 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
 	if t.TLSHandshakeTimeout == 0 {
 		t.TLSHandshakeTimeout = defaultTransport.TLSHandshakeTimeout
 	}
+	if err := http2.ConfigureTransport(t); err != nil {
```

Comment:
I only noticed it because I saw the comment above the line you pasted and
it intrigued me :)

We have the TLS transport cache so in practice this won't cause a
performance issue.  We just need to be sure we're aware of any wrinkles.

On Mon, May 9, 2016 at 1:35 PM, krousey <notifications@github.com> wrote:

> In pkg/util/net/http.go
> <https://github.com/kubernetes/kubernetes/pull/25280#discussion_r62539066>
> :
>
> > @@ -67,6 +70,9 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
> >  	if t.TLSHandshakeTimeout == 0 {
> >  		t.TLSHandshakeTimeout = defaultTransport.TLSHandshakeTimeout
> >  	}
> > +	if err := http2.ConfigureTransport(t); err != nil {
>
> Picking through the upstream code, I now see why this has to be done.
> Thanks!
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25280/files/53ddc93c6e80e750d09fa72680c718754e0693bc#r62539066>
>